src/java/com/puppetlabs/http/client/Sync.java

```diff
@@ -57,7 +57,7 @@ private static ClientOptions extractClientOptions(SimpleRequestOptions simpleOpt
         return new ClientOptions(sslContext, sslCert, sslKey, sslCaCert,
                 sslProtocols, sslCipherSuites, insecure,
                 forceRedirects, followRedirects, connectTimeoutMilliseconds,
-                socketTimeoutMilliseconds);
+                socketTimeoutMilliseconds, 1, 1);
```

Comment:
Should these be the defaults of 2 and 20?

Reply:
Or 0 0?

Reply:
I struggled a bit to find a good default.  This is the simple "one shot" client config, so it will at most ever be one connection before it is closed.  I thought making that explicit might be a good idea.  I could add comments explaining why they are 1,1 or choose a diff default.  The values don't impact resource allocation, just limits.

Reply:
That makes a lot of sense. Let's document the reason behind these values and *not* put this into Puppet 6/Johnson until next week. Then we have a bit of runway to see if there's any weird interactions when we plug this into Puppet before we do a release.